standard/expressions.md

```diff
@@ -1512,7 +1512,14 @@ tuple_element
     : (identifier ':')? expression
 ```
 
-A tuple expression is classified as a tuple. It has a type if all the element expressions have a type. In that case, the type of a tuple expression is the tuple type of the same arity, where each type element has the type - and name, where given - of the corresponding element expression.
+A tuple expression is classified as a tuple. A tuple expression has a type if and only if each of its element expressions `Ei` has a type `Ti`. The type shall be a tuple type of the same arity as the tuple expression, where each element is given by the following:
+
+- If the tuple element in the corresponding position has a name `Ii`, then the tuple type element shall be `Ti Ii`.
```

Comment:
Is there some other letter we could pick instead of `I` here? Maybe `Ni` instead of `Ii` (captain)?

Reply:
Fair point. Changed to `Ni` in upcoming commit.